File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Projects/updateprojectfile.rb
module Asposetasksjava
  module UpdateProjectFile
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        # create a new task
        task = project.getRootTask().getChildren().add("Task1")
        
        # set start date
        cal = Rjb::import('java.util.Calendar').getInstance()

        cal.set(2015, 7, 1, 8, 0,0);
        task.set(Rjb::import('com.aspose.tasks.Tsk').START, cal.getTime())

        cal.set(2015, 7, 1, 17, 0, 0)
        task.set(Rjb::import('com.aspose.tasks.Tsk').FINISH , cal.getTime())

        # Save the project as MPP project file
        project.save(data_dir + "AfterLinking.mpp", Rjb::import('com.aspose.tasks.SaveFileFormat').MPP)

        puts "Project file updated, please check the output file."
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithProjects/UpdateProjectFile.php
<?php
namespace Aspose\Tasks\WorkingWithProjects;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\Tsk as Tsk;
use com\aspose\tasks\SaveFileFormat as SaveFileFormat;

use java\util\Calendar as Calendar;
class UpdateProjectFile{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');
        # create a new task
        $task = $project->getRootTask()->getChildren()->add("Task1");

        # set start date
        $cal=new Calendar();
        $cal = $cal->getInstance();

        $cal->set(2015, 7, 1, 8, 0,0);

        $tsk=new Tsk();
        $task->set($tsk->START, $cal->getTime());

        $cal->set(2015, 7, 1, 17, 0, 0);
        $task->set($tsk->FINISH , $cal->getTime());

        # Save the project as MPP project file
        $saveFileFormat=new SaveFileFormat();
        $project->save($dataDir . "AfterLinking.mpp", $saveFileFormat->MPP);
        print "Project file updated, please check the output file.".PHP_EOL;
    }

}
?>